src/components/ProductCard.jsx

```diff
@@ -1,5 +1,16 @@
+import { useNavigate } from "react-router-dom";
+import { useCart } from "../contexts";
+
 export const ProductCard = ({ productDetails }) => {
-  const { title, price, discount, discountPrice, imageUrl } = productDetails;
+  const { _id, title, price, discount, discountPrice, imageUrl } =
+    productDetails;
+  const {
+    cartState: { cart, wishlist },
+    cartDispatch,
+  } = useCart();
+  const navigate = useNavigate();
+  const productInCart = cart.find((product) => product._id === _id);
+  const productInWishlist = wishlist.find((product) => product._id === _id);
```

Comment:
These functions have same logic so you can extract it out to a common utility function

Reply:
Done